Userland/Libraries/LibWeb/WebDriver/ExecuteScript.cpp

```diff
@@ -395,12 +395,22 @@ ExecuteScriptResultSerialized execute_async_script(Web::Page& page, DeprecatedSt
             WebIDL::reject_promise(realm, promise_capability, script_promise.result());
     }();
 
-    // FIXME: 6. If promise is still pending and session script timeout milliseconds is reached, return error with error code script timeout.
-
+    // 6. If promise is still pending and session script timeout milliseconds is reached, return error with error code script timeout.
     vm.custom_data()->spin_event_loop_until([&] {
+        if (timeout.has_value() && (MonotonicTime::now() - start) > Duration::from_seconds(static_cast<i64>(*timeout))) {
+            return true;
+        }
+
         return promise->state() != JS::Promise::State::Pending;
     });
 
+    if (timeout.has_value() && (MonotonicTime::now() - start) > Duration::from_seconds(static_cast<i64>(*timeout))) {
```

Comment:
This long `if` statement being copy-pasted a bunch of times doesn't feel great (note it also exists in the IIFE above) , can we throw it in a lamdba?
```c++
auto has_timed_out = [&]() {
    return timeout.has_value() && (MonotonicTime::now() - start) > Duration::from_seconds(static_cast<i64>(*timeout));
};
```